### DISCUSSION TITLE
Treesitter?

### DISCUSSION THREAD

**Discussion by ghost** [supabase-community/postgres-language-server#29] [Q&A]
Hey,

Reading through the readme I noticed _Semantic Highlighting_, will that be done through treesitter?

**Comment by psteinroe** [comment]
hey @unansweredocd,

the semantic highlighting in the readme refers to the [Semantic Tokens](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_semanticTokens) spec of the language server protocol. 

But a tree sitter plugin might come in handy for supporting `pl/rust` and `pl/node` functions. client-side plugins such as [otter](https://github.com/jmbuhr/otter.nvim) rely on tree-sitter to provide lsp features for code embedded in other code or documents.

I have never worked with the tree sitter syntax, buy it might be possible to generate a definition from the `proto` definition of libg_query.

**Reply by ghost** [reply]
I see, reason I asked is because much like an lsp, there are only general sql treesitter options available and none of em' quite get postgres right. I suppose that's something entirely different compared to this lsp so I'll close the discussion as my question was answered, thank you @psteinroe

**Reply by psteinroe** [reply]
It's definitely on my wishlist for this project, too! Will come back to this once the basics of the lsp are done.
